### DISCUSSION TITLE
Customize language code on translate output files?

### DISCUSSION THREAD

**Discussion by HarukaMa** [WeblateOrg/weblate#5666] [General]
It seems that Weblate is using(?) RFC style language codes and while it has some built-in aliases, when exporting the translations, they are always using "main" language code, which may not correctly match the language codes used in different platforms.

For example, Weblate uses `zh_Hans` as language code of Simplified Chinese, so the output file could be `zh_Hans.po`. However platforms may be using code `zh_CN`, and when searching for language files, `zh_CN.po` will be missing. We have to rely on some fallback features (the library searches for `zh.po` then `zh_*.po` instead).

I'm aware that I can edit existing language definitions or even add new languages, but are there better solutions than that?

**Comment by nijel** [comment]
The best solution is to read the documentation on this topic: https://docs.weblate.org/en/latest/admin/languages.html#adding-new-translations

**Reply by HarukaMa** [reply]
Currently I have `en_US:en,ja_JP:ja,zh_CN:zh_Hans`, but after adding Japanese and Simplified Chinese translations, when downloading translations I still have `ja` and `zh_Hans` as filename. 

I really can't see anything about it in the linked doc except:

> Additionally, any mappings defined in Language aliases are applied in reverse.

However it seems this is not applied when downloading translations.

From https://docs.weblate.org/en/latest/admin/projects.html#project-language-aliases, it seems the alias is actually in the correct order:

> Using non standard code: ia_FOO:ia

as the desired language code should be on the left side of the colon.

Workflow:

1. Create a zip containing only `strings.pot` as initial translation files
2. Create a component using the zip
3. Use gettext PO format and `strings.pot` as the template of new translations
4. Add target language as new translations

**Reply by HarukaMa** [reply]
It seems the language codes are actually correct in the weblate repo, but not when downloading files directly from web page.

**Reply by nijel** [reply]
The URLs always use languages as defined in Weblate.
